src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.cs

```diff
@@ -0,0 +1,694 @@
+        [|Obsoleted()|];
+        [|Removed()|];
+    }
+    [MinimumOSPlatform(""Windows10.1.1.1"")]
+    public void WindowsOnly()
+    {
+    }
+    [ObsoletedInOSPlatform(""Linux4.1"")]
+    public void Obsoleted()
+    {
+    }
+    [RemovedInOSPlatform(""Linux4.1"")]
+    public void Removed()
+    {
+    }
+}
+" + MockAttributesSource;
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task WrongPlatformStringsShouldHandledGracefully()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    public void M1()
+    {
+        Windows10();
+        Windows1_2_3_4_5();
+        [|ObsoletedLinuxDash4_1()|];
+        [|ObsoletedLinuxStar4_1()|];
+        [|RemovedLinu4_1()|];
+        ObsoletedWithNullString();
+        RemovedWithEmptyString();
+    }
+    [MinimumOSPlatform(""Windows10"")]
+    public void Windows10()
+    {
+    }
+    [MinimumOSPlatform(""Windows1.2.3.4.5"")]
+    public void Windows1_2_3_4_5()
+    {
+    }
+    [ObsoletedInOSPlatform(""Linux-4.1"")]
+    public void ObsoletedLinuxDash4_1()
+    {
+    }
+    [ObsoletedInOSPlatform(""Linux*4.1"")]
+    public void ObsoletedLinuxStar4_1()
+    {
+    }
+    [ObsoletedInOSPlatform(null)]
+    public void ObsoletedWithNullString()
+    {
+    }
+    [RemovedInOSPlatform(""Linu4.1"")]
+    public void RemovedLinu4_1()
+    {
+    }
+    [RemovedInOSPlatform("""")]
+    public void RemovedWithEmptyString()
+    {
+    }
+}
+" + MockAttributesSource;
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task OsDependentPropertyCalledWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [MinimumOSPlatform(""Windows10.1.1"")]
+    public string WindowsStringProperty { get; set; }
+    [ObsoletedInOSPlatform(""ios4.1"")]
+    public int ObsoleteIntProperty { get; set; }
+    [RemovedInOSPlatform(""Linux4.1"")]
+    public byte RemovedProperty { get; }
+    public void M1()
+    {
+        [|WindowsStringProperty|] = ""Hello"";
+        string s = [|WindowsStringProperty|];
+        M2([|WindowsStringProperty|]);
+        [|ObsoleteIntProperty|] = 5;
+        M3([|ObsoleteIntProperty|]);
+        M3([|RemovedProperty|]);
+    }
+    public string M2(string option)
+    {
+        return option;
+    }
+    public int M3(int option)
+    {
+        return option;
+    }
+}
+" + MockAttributesSource;
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Theory]
+        [InlineData("MinimumOSPlatform", "string StringProperty", " == StringProperty", @"StringProperty|] = ""Hello""", "StringProperty")]
+        [InlineData("ObsoletedInOSPlatform", "int IntProperty", " > IntProperty", "IntProperty|] = 5", "IntProperty")]
+        [InlineData("RemovedInOSPlatform", "int RemovedProperty", " <= RemovedProperty", "RemovedProperty|] = 3", "RemovedProperty")]
```

Comment:
You could convert these into an `IEnumerable<object[]>` and iterate through all the values. Makes it a bit more readable when you have to test many combinations with multiple lists of repeated values. Example: the [IEnumerable](https://github.com/dotnet/runtime/blob/cb1a7feaa43cdbd03b317db24fa30d86eae2a27c/src/libraries/System.IO.Pipes.AccessControl/tests/AnonymousPipeTests/AnonymousPipeServerStreamAclTests.cs#L72-L75) and the [MemberData](https://github.com/dotnet/runtime/blob/cb1a7feaa43cdbd03b317db24fa30d86eae2a27c/src/libraries/System.IO.Pipes.AccessControl/tests/AnonymousPipeTests/AnonymousPipeServerStreamAclTests.cs#L79).

Reply:
yeah that might look cleaner